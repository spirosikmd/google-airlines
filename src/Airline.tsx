import React from 'react';
import Image from 'react-graceful-image';
import { AirlineData } from './airlines';

const Airline = ({ airline }: { airline: AirlineData }) => (
  <div className="app__airline" data-tip={airline.airline} tabIndex={0}>
    <Image
      src={`//www.gstatic.com/flights/airline_logos/70px/${airline.iata}.png`}
      height="35"
      width="35"
      alt={airline.airline}
      placeholderColor="#E5FCF5"
      noRetry={true}
    />
  </div>
);

export default Airline;
