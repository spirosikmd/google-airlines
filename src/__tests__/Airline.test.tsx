import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Airline from '../Airline';

afterEach(cleanup);

test('renders an airline with image', () => {
  const airline = { id: 'id', iata: 'iata', airline: 'airline' };
  const { asFragment } = render(<Airline airline={airline} />);

  expect(asFragment()).toMatchSnapshot();
});
