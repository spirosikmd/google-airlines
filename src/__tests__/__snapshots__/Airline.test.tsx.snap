// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders an airline with image 1`] = `
<DocumentFragment>
  <div
    class="p-6 m-2 border border-solid rounded-full border-gray-200  hover:border-purple-600"
    data-tip="airline"
    tabindex="0"
  >
    <img
      alt="airline"
      height="35"
      src="data:image/svg+xml;charset=utf-8,%3Csvg xmlns%3D'http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg' width%3D'35' height%3D'35' viewBox%3D'0 0 35 35'%2F%3E"
      style="background: rgb(229, 252, 245);"
      width="35"
    />
  </div>
</DocumentFragment>
`;
